openpaas-james/jmap/extensions/src/main/scala/com/linagora/openpaas/james/jmap/method/FilterSetMethod.scala

```diff
@@ -61,7 +62,13 @@ case class FilterSetUpdateFailure(id: String, exception: Throwable) extends Filt
   override def notUpdated: Map[String, FilterSetError] = Map(id -> asSetError(exception))
 
   def asSetError(exception: Throwable): FilterSetError = exception match {
-    case e: IllegalArgumentException => FilterSetError.invalidArgument(Some(SetErrorDescription(e.getMessage)))
+    case e: IllegalArgumentException => {
+      if (e.getMessage.equals("Provided state must be as same as the current state")) {
```

Comment:
Can you introduce a specific exception type so that we do not need to lookup the message?|

Then we can write:

```
def asSetError(exception: Throwable): FilterSetError = exception match {
    case e: IllegalArgumentException => FilterSetError.invalidArgument(Some(SetErrorDescription(e.getMessage)))
    case e: StateMismatchException => FilterSetError.stateMismatch(Some(SetErrorDescription(e.getMessage)))
}
```

Reply:
Maybe the Apache FilterManagement stuff should be throwing an IllegalStateException?